app/src/androidTest/java/org/mozilla/focus/activity/FindInPageTest.kt

```diff
@@ -0,0 +1,255 @@
+
+package org.mozilla.focus.activity
+
+import android.content.Intent
+import android.net.Uri
+import android.support.annotation.Keep
+import android.support.test.InstrumentationRegistry
+import android.support.test.espresso.Espresso
+import android.support.test.espresso.Espresso.onView
+import android.support.test.espresso.IdlingRegistry
+import android.support.test.espresso.action.ViewActions.click
+import android.support.test.espresso.action.ViewActions.swipeDown
+import android.support.test.espresso.action.ViewActions.swipeUp
+import android.support.test.rule.ActivityTestRule
+import android.support.test.runner.AndroidJUnit4
+import android.support.test.espresso.assertion.ViewAssertions.matches
+import android.support.test.espresso.matcher.ViewMatchers
+import android.support.test.espresso.matcher.ViewMatchers.isDisplayed
+import android.support.test.espresso.matcher.ViewMatchers.withId
+import android.support.test.espresso.matcher.ViewMatchers.withText
+import android.support.test.espresso.matcher.ViewMatchers.withEffectiveVisibility
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.focus.R
+import org.mozilla.focus.autobot.findInPage
+import org.mozilla.focus.autobot.session
+import org.mozilla.focus.helper.BeforeTestTask
+import org.mozilla.focus.helper.SessionLoadedIdlingResource
+
+@Keep
+@RunWith(AndroidJUnit4::class)
+class FindInPageTest {
+
+    @JvmField
+    @Rule
+    val activityTestRule = ActivityTestRule(MainActivity::class.java, true, false)
+
+    @Before
+    fun setUp() {
+        BeforeTestTask.Builder()
+                .enableSreenshotOnBoarding(true)
+                .build()
+                .execute()
+        activityTestRule.launchActivity(Intent())
+    }
+
+    companion object {
+        private const val TARGET_URL_SITE_1 = "file:///android_asset/gpl.html"
+        private const val TARGET_URL_SITE_2 = "https://developer.mozilla.org/en-US/Firefox_for_Android"
+        private const val keyword = "program"
+    }
+
+    /**
+     * Test case no: TC0147
+     * Test case name: Find a specific word or phrase on a page
+     * Steps
+     * 1. Launch Rocket
+     * 2. Visit mozilla.org
+     * 3. Tap Menu -> Find in page
+     * 4. Find a specific word or phrase
+     * 5. Scroll up and down
+     */
+    /**
+     * Test case no: TC0154
+     * Test case name: Check mini url bar and the bottom action in "Find in page" mode
+     * 1. Launch Rocket
+     * 2. Login facebook.com with your account
+     * 3. Tap Menu -> Find in page
+     * 4. Find the word
+     * 5. Scroll up and down
+     */
+    @Test
+    fun findInPageAndScroll() {
+
+        session {
+            loadPageFromHomeSearchField(activityTestRule.activity, TARGET_URL_SITE_1)
+            clickBrowserMenu()
+            clickMenuFindInPage()
+        }
+
+        findInPage {
+            findKeywordInPage(keyword)
+
```

Comment:
Please move below onView to a robot.
In case we'll have some refactoring around webview_slot and needed to refactor the id everywhere.